### PR TITLE
Add IO#read_string?

### PR DESCRIPTION
**Pull request by straight-shoota**
Fixes part of #6905

### CODE REVIEW

File: spec/std/io/io_spec.cr [hunk 1]
```diff
@@ -333,6 +333,14 @@ describe IO do
       end
     end
 
+    it "#read_string?" do
+      io = SimpleIOMemory.new("hello world")
+      io.read_string?(5).should eq("hello")
+      io.read_string?(1).should eq(" ")
+      io.read_string?(0).should eq("")
```
Comment (luislavena) [inline spec/std/io/io_spec.cr]: This scenario is not described in the documentation. I know originally isn't described for `#read_string` either, but might be worth adding it.

Thank you.

File: src/io.cr [hunk 2]
```diff
@@ -443,14 +443,30 @@ abstract class IO
...
     String.new(bytesize) do |ptr|
       if decoder = decoder()
         read = decoder.read_utf8(self, Slice.new(ptr, bytesize))
         if read != bytesize
-          raise IO::EOFError.new
+          return
         end
       else
-        read_fully(Slice.new(ptr, bytesize))
+        unless read_fully?(Slice.new(ptr, bytesize))
```
Comment (j8r) [inline src/io.cr]: Why you don't put this in an `elif`?

**Comment by jhass** [comment]
Mh, sorry to derail a bit, but this made me realize it makes it easy to read a broken string if your bytelimit happens to be in the middle of a multi-byte character (whatever multi-byte encoding your IO is in really). I wonder if these should take the amount of codepoints/characters to read instead?

**Comment by asterite** [comment]
I'm actually thinking this method shouldn't exist. The problem is that it can happen that you read some bytes but not enough to reach the desired bytesize, and at that point you get `nil` as a return value but you lost those read bytes.

With `read_fully?` at least you get the bytes on the passed slice so you don't lose them.

Or, well, at least I would document that this method might read bytes that will be lost, and most probably the caller should raise an error on `nil`.

**Comment by RX14** [comment]
I also want `gets(limit)` without stopping at any delimiters. gets's behaviour is to only return nil if the current position is EOF, it returns strings whose bytesize is less than the limit. So ideally there should be both behaviours. The behaviour in this PR seems more like `read_string_fully` to me, and we need a `read_string`.

**Comment by straight-shoota** [comment]
Closing stale PR.

**Pull request closed without merging**